src/core/thread/mle_router.cpp

```diff
@@ -3866,6 +3869,17 @@ void MleRouter::HandleAddressSolicitResponse(Coap::Message *         aMessage,
         leader->SetNextHop(RouterIdFromRloc16(mParent.GetRloc16()));
     }
 
+    if (isRouterIdAllocated)
+    {
+        // send Link Request immediately if Router ID was previously allocated
+        IgnoreError(SendLinkRequest(nullptr));
+    }
+    else
+    {
+        // wait to send Link Request until new Router ID has been disseminated from the Leader
+        mLinkRequestDelay = kMulticastLinkRequestDelay;
```

Comment:
Can we consider removing/optimizing this? We set `mLinkRequestDelay = kMulticastLinkRequestDelay` few lines below?

Reply:
Thanks for catching that. Updated.